### ISSUE TITLE
Handling Input Creation for FieldResolvers!

### ISSUE THREAD

**Issue by nwmartin** [goldcaddy77/warthog#271]
Heyo,

Right now non-database backed fields won't get where inputs generated for them for something like a FieldResolver:

```
  @FieldResolver(() => String)
  async status(@Root() entity: Entity) {
    return "JAY ITS A SEA MONSTAH"
  }
```

Right now we're just extending the input sets to include them as possibilities for where queries, but it would be nice if they were potentially auto-generated or if slapped a Field decorator on them or something.

Let me know if you need more details.

**Comment by goldcaddy77** [comment]
Interesting... can you give me a more concrete example?  Field Resolvers typically reach into an association.  For instance, the `User` model would have a `FieldResolver` for `posts`, that would load up all the posts.  I'm not sure what the where inputs would do there.  If you wanted to expose a field just to the GraphQL layer, we can make the decorators smarter, perhaps by adding the possibility to add `{ db: false }`, then it would do everything it normally does, but skips the TypeORM decorator.

**Comment by nwmartin** [comment]
You're right in that we might be reaching into an association, but imagine a case where what is actually resolved is a single entity (the latest one, perhaps). You end up with a case where the resolved item, from the point of view of the user, is literally just a simple field. We don't automatically get any inputs around this without a lot of manual effort.

Our current workaround was to add in the inputs we wanted (`eq` and `in`) and then needed to use a typeorm query builder with custom where clause generation to actually get data back that used the underlying where clause. It's a lot of lifting. There might be a better way of doing this - I can show you this example outside of this issue.
